test/functional/rpc_rawtransaction.py

```diff
@@ -62,44 +64,43 @@ def set_test_params(self):
         # whitelist all peers to speed up tx relay / mempool sync
         for args in self.extra_args:
             args.append("-whitelist=noban@127.0.0.1")
+        self.requires_wallet = self.is_wallet_compiled()
 
         self.supports_cli = False
 
     def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
+        if self.requires_wallet:
+            if self.options.descriptors:
+                self.skip_if_no_sqlite()
+            else:
+                self.skip_if_no_bdb()
 
     def setup_network(self):
         super().setup_network()
         self.connect_nodes(0, 2)
 
     def run_test(self):
+        self.wallet = MiniWallet(self.nodes[0])
         self.log.info("Prepare some coins for multiple *rawtransaction commands")
-        self.generate(self.nodes[2], 1)
+        self.generate(self.wallet, 10)
         self.generate(self.nodes[0], COINBASE_MATURITY + 1)
-        for amount in [1.5, 1.0, 5.0]:
-            self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), amount)
-        self.sync_all()
-        self.generate(self.nodes[0], 5)
 
         self.getrawtransaction_tests()
         self.createrawtransaction_tests()
-        self.signrawtransactionwithwallet_tests()
+        if self.requires_wallet:
+            self.signrawtransactionwithwallet_tests()
         self.sendrawtransaction_tests()
         self.sendrawtransaction_testmempoolaccept_tests()
         self.decoderawtransaction_tests()
         self.transaction_version_number_tests()
-        if not self.options.descriptors:
+        if self.requires_wallet and not self.options.descriptors:
             self.raw_multisig_transaction_legacy_tests()
 
     def getrawtransaction_tests(self):
-        addr = self.nodes[1].getnewaddress()
-        txid = self.nodes[0].sendtoaddress(addr, 10)
+        tx = self.wallet.send_self_transfer(from_node=self.nodes[1])
+        self.sync_all()
         self.generate(self.nodes[0], 1)
-        vout = find_vout_for_address(self.nodes[1], txid, addr)
-        rawTx = self.nodes[1].createrawtransaction([{'txid': txid, 'vout': vout}], {self.nodes[1].getnewaddress(): 9.999})
-        rawTxSigned = self.nodes[1].signrawtransactionwithwallet(rawTx)
-        txId = self.nodes[1].sendrawtransaction(rawTxSigned['hex'])
-        self.generateblock(self.nodes[0], output=self.nodes[0].getnewaddress(), transactions=[rawTxSigned['hex']])
+        txId = tx["txid"]
```

Comment:
nit, consistency with your changes
```suggestion
        txId = tx['txid']
```